Dockerfile

```diff
@@ -138,12 +138,44 @@ ENV LD_LIBRARY_PATH=/usr/local/lib
 ENV PATH=/opt/hadoop/libexec:$PATH:/opt/hadoop/bin
 
 RUN groupadd --gid 1000 hadoop
-RUN useradd --uid 1000 hadoop --gid 100 --home /opt/hadoop
+RUN useradd --uid 1000 hadoop --gid 1000 --home /opt/hadoop
 RUN chmod 755 /opt/hadoop
 RUN echo "hadoop ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
 
 RUN chown hadoop /opt
 
+RUN groupadd --gid 1001 om
+RUN useradd --uid 1001 om --gid 1001 --home /opt/om
+RUN chmod 755 /opt/om
+
+RUN groupadd --gid 1002 dn
+RUN useradd --uid 1002 dn --gid 1002 --home /opt/dn
+RUN chmod 755 /opt/dn
+
+RUN groupadd --gid 1003 scm
+RUN useradd --uid 1003 scm --gid 1003 --home /opt/scm
+RUN chmod 755 /opt/scm
+
+RUN groupadd --gid 1004 s3g
+RUN useradd --uid 1004 s3g --gid 1004 --home /opt/s3g
+RUN chmod 755 /opt/s3g
+
+RUN groupadd --gid 1005 rm
+RUN useradd --uid 1005 rm --gid 1005 --home /opt/rm
+RUN chmod 755 /opt/rm
```

Comment:
Why do we need user and home for `rm`?  AFAIK it's a Hadoop component and it runs in Hadoop image.

Reply:
Makes sense.